test/90_1_general/test_179d_building_type.rb

```diff
@@ -0,0 +1,274 @@
+    assert_equal(standards_space_type, data['space_type'])
+    assert_equal('ASHRAE 90.1-2007', data['lighting_standard'])
+    assert_equal('WholeBuilding', data['lighting_primary_space_type'])
+    assert_equal(lighting_secondary_space_type, data['lighting_secondary_space_type'])
+    assert_in_delta(expected_lpd_ip, data['lighting_per_area'])
+    assert_equal(lighting_schedule, data['lighting_schedule'])
+  end
+
+  OFFICE_DATA.each do |building_area_sqft, n_floors, building_type, expected_primary_building_type, standards_space_type|
+    define_method("test_office_get_building_type_#{building_area_sqft.to_i}sqft_#{n_floors}floors") do
+      building_type_test(building_area_sqft, n_floors, building_type, expected_primary_building_type, standards_space_type, OFFICE_LPD, 'Office', 'Nonres_Light_Sch')
+    end
+  end
+
+  def test_get_building_type_sanitizes
+    building_area_sqft, n_floors, building_type, expected_primary_building_type, standards_space_type = [160_000, 5, 'Office', 'LargeOffice', 'WholeBuilding - Lg Office']
+    model = make_rectangular_building_with_area(building_area_sqft, n_floors)
+    assert_equal(5, model.getSpaces.size)
+
+    spaces = model.getSpaces.sort_by(&:nameString)
+
+    building_types = ['SmallOffice', 'MediumOffice', 'LargeOffice']
+    building_types.zip(spaces).each do |building_type, s|
+      z = OpenStudio::Model::ThermalZone.new(model)
+      s.setThermalZone(z)
+
+      sp = OpenStudio::Model::SpaceType.new(model)
+      sp.setStandardsBuildingType(building_type)
+      sp.setStandardsSpaceType('Gibberish')
+      s.setSpaceType(sp)
+    end
+
+    sp_warehouse = sp = OpenStudio::Model::SpaceType.new(model)
+    sp_warehouse.setStandardsBuildingType('Warehouse')
+    sp_warehouse.setStandardsSpaceType('Gibberish')
+    spaces[-2..].each do |s|
+      s.setSpaceType(sp_warehouse)
+    end
+
+    model.getBuilding.setStandardsBuildingType('FullServiceRestaurant')
+
+    # We use model_get_lookup_name inside the model_get_primary_building_type,
+    # so the first two resolve to 'Office' and it 'wins' the space_area method
+    # And the space area method trumps the building one!
+    primary_building_type_no_remap = @standard.model_get_primary_building_type(model, remap_office: false)
+    assert_equal(building_type, primary_building_type_no_remap)
+
+    primary_building_type = @standard.model_get_primary_building_type(model, remap_office: true)
+    assert_equal(expected_primary_building_type, primary_building_type)
+
+    sp_type_name = @standard.whole_building_space_type_name(model, 'Office')
+    assert_equal(standards_space_type, sp_type_name)
+
+    warns =  @@logSink.logMessages.select { |l| l.logLevel == OpenStudio::Warn }
+    assert_equal(1, warns.size)
+    logMessage = warns.first.logMessage
+    assert_equal("The Building has standardsBuildingType 'FullServiceRestaurant' while the area determination based on space types has 'Office'. Preferring the Space Type one", logMessage)
+  end
```

Comment:
See:
* 25% of the building has a space type with a standards building type of 'SmallOffice'
* 25% MediumOffice
* 25% LargeOffice
* 25% Warehouse

Still, 'Office' is picked up at 75%, and selected. Then I can remap_office: true to get 'LargeOffice' because it's bigger than 150_000 sqft